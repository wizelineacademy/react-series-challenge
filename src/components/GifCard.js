import React from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';

import favoriteActions from '../actions/favoriteActions';

import emptyStar from '../icons/empty-star.svg';
import star from '../icons/star.svg';

import {
  GifCardWrapper,
  GifCardStar,
  GifCardMainImage
} from '../styles/GifCard.style.js';

export const GifCard = (props) => {
  const { favorited, id, images, title, toggleFavorite } = props;

  return (
    <GifCardWrapper>
      <GifCardStar
        src={ favorited ? star : emptyStar }
        alt="favorite icon"
        onClick={ () => toggleFavorite({ id, title, url: images.downsized.url }) }
      />

      <GifCardMainImage src={ images.downsized.url } alt={title} />
    </GifCardWrapper>
  );
};

const mapStateToProps = (state, props) => {
  const { id } = props;
  const { favorites } = state;

  return {
    favorited: !!favorites.find((fav) => fav.id === id)
  };
};

const mapDispatchToProps = (dispatch) => {
  const { toggleFavorite } = favoriteActions.creators;

  return bindActionCreators({ toggleFavorite }, dispatch);
};

export default connect(mapStateToProps, mapDispatchToProps)(GifCard);
