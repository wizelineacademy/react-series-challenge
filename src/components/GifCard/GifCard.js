import React, {Component} from 'react';
import {bindActionCreators} from "redux";
import { connect } from 'react-redux';
import {
    GifContainer,
    Gif,
    GifOverlay,
    GifTitle,
    GifOverlayLoading,
    FavoriteBtn
} from './GifCard.styled';
import favoriteGifsActions from "../../actions/favoriteGifs";

const LOCAL_STORAGE_FAV_GIFS = 'favoriteGifs';

class GifCard extends Component {

    constructor(props) {
        super(props);

        let isFavorite = this.gifInLocalStorage();
        if (isFavorite) {
            this.props.addFavoriteGif({ gif: this.props.gif, isFavorite });
        }
        this.state = { gifLoaded: false, isFavorite };
    }

    gifInLocalStorage() {
        const localFavoriteGifs = localStorage.getItem(LOCAL_STORAGE_FAV_GIFS);
        const jsonGifs = JSON.parse(localFavoriteGifs);
        return (jsonGifs && !!jsonGifs[this.props.gif.id]);
    }

    onLoadHandler = () => {
        const gifLoaded = !this.state.gifLoaded;
        this.setState({ gifLoaded });
    };

    onClickFavoriteButton = () => {

        const isFavorite = !this.state.isFavorite;
        const { gif } = this.props;
        this.setState({ isFavorite });

        if (isFavorite) {
            this.props.addFavoriteGif({ gif, isFavorite });
        } else {
            this.props.deleteFavoriteGif({ gif, isFavorite });
        }
    };

    render() {

        const { gif } = this.props;

        const buttonIcon = this.state.isFavorite ? <i className="fas fa-heart" /> : <i className="far fa-heart" />;

        const gifOverlay = (
            <GifOverlay>
                <GifTitle>
                    {gif.title || ''}
                </GifTitle>
                <FavoriteBtn onClick={this.onClickFavoriteButton} isFavorite={this.state.isFavorite}>
                    { buttonIcon }
                </FavoriteBtn>
            </GifOverlay>
        );

        const gifOverlayLoading = (
            <GifOverlayLoading>
                <p>Loading...</p>
                <p>{gif.title}</p>
            </GifOverlayLoading>
        );

        return (
            <GifContainer>
                <Gif src={gif.images.original.webp} onLoad={this.onLoadHandler} gifLoaded={this.state.gifLoaded ? 'LOADED' : 'NOT_LOADED'}/>
                {!this.state.gifLoaded ? gifOverlayLoading : gifOverlay}
            </GifContainer>
        );
    }

}


const mapDispatchToProps = (dispatch) => {

    const { addFavoriteGif, deleteFavoriteGif } = favoriteGifsActions.creators;

    return bindActionCreators({
        addFavoriteGif,
        deleteFavoriteGif,
    }, dispatch);

};

export default connect(null, mapDispatchToProps)(GifCard);
