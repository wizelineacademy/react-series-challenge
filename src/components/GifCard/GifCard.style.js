import styled from 'styled-components';

export const BtnsWrapper =  styled.div`
  align-items: center;
  background: rgba(255, 255, 255, 0.7);
  bottom: 0;
  color: #fff;
  display: flex;
  justify-content: center;
  left: 0;
  opacity: 0;
  position: absolute;
  right: 0;
  transition: opacity .2s, visibility .2s;
  top: 0;
  visibility: hidden;
`;

export const GifCardWrapper = styled.div`
  padding: 3px;
  position: relative;
  width: fit-content;
  
  &:hover ${BtnsWrapper} {
    opacity: 1;
    visibility: visible;
  }
`;

export const Btn = styled.img`
  cursor: pointer;
  height: 45px;
  padding: 0 15px;
  width: 45px;
`;

Btn.displayName = 'Btn';
