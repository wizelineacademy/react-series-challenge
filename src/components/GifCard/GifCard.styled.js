import styled from 'styled-components';

const GifContainer = styled.div`
    position: relative;
`;

const GifOverlay = styled.div`
    position: absolute;
    top: 0;
    left: 0;
    bottom: 0;
    right: 0;
    color: #fff;
    background-color: rgba(0,0,0,0.0);
    transition: background-color 0.2s ease;
    
    &:hover {
        background-color: rgba(0,0,0,0.7);
    }
`;

const GifTitle = styled.div`
    visibility: hidden;
    font-size: 1rem;
    padding: 8px;
    
    ${GifOverlay}:hover & {
        visibility: visible;
        opacity: 1;
    }
`;

const GifOverlayLoading = styled.div`
    position: absolute;
    top: 0;
    left: 0;
    bottom: 0;
    right: 0;
    color: #fff;
    background-color: rgba(0,0,0,0.9);
    text-align: center;
`;

const Gif = styled.img`
    display: block;
    border: 0;
    width: 100%;
    height: 150px;
    max-height: 150px;
    // height: auto;
`;

const FavoriteBtn = styled.button`
    position: absolute;
    visibility: hidden;
    bottom: 1rem;
    right: 1rem;
    font-size: 1.4rem;
    background: transparent;
    border: none;
    color: ${(props) => props.isFavorite ? '#E93D44' : '#FFF'};
    cursor: pointer;
    
    &:focus {
        outline: none !important;
    }
    
    ${GifOverlay}:hover & {
        visibility: visible;
        opacity: 1;
    }
`;

export {
    GifContainer,
    GifOverlay,
    GifTitle,
    GifOverlayLoading,
    Gif,
    FavoriteBtn,
};