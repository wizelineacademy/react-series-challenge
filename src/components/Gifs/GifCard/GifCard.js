import React, { Component } from 'react'
import { connect } from 'react-redux'
import * as actionTypes from '../../../actions'

import {
  GifCardWrapper,
  GifCardItem,
  GifImage,
  GifCardFavorites,
  FavoriteLogo
} from './GifCard.styled'

class GifCard extends Component {
  constructor(props) {
    super(props)

    this.state = {
      isFavorite: false
    }
  }

  componentDidMount() {
    const favoriteGifs = JSON.parse(localStorage.getItem('favoriteGifs'))
    const { gif } = this.props

    const checkIndex = favoriteGifs.findIndex(element => element.id === gif.id)

    if (checkIndex > -1) {
      this.setState({
        isFavorite: true
      })
    }
  }

  handleToogleFavorite= () => {
    const { gif, onToggleFavorite } = this.props

    onToggleFavorite(gif)

    this.setState( prevState => ({
      isFavorite: !prevState.isFavorite
    }));

  }

  render() {
    const { gif } = this.props
    const { isFavorite } = this.state

    return (
      <GifCardWrapper className="GifCard">
        <GifCardItem href="#" className="GifCardItem">
          <GifImage
            className="GifImage"
            src={gif.images.downsized.url}
            alt={gif.title}
          />
        </GifCardItem>
        <GifCardFavorites className="GifCardFavorites">
          <FavoriteLogo
            size="20"
            // onClick={() => onToggleFavorite(gif)}
            onClick={this.handleToogleFavorite}
            isFavorite={isFavorite}
          />
        </GifCardFavorites>
      </GifCardWrapper>
    )
  }
}


const mapDispatchToProps = dispatch => {
  return {
    onToggleFavorite: item => dispatch(actionTypes.favoriteToggleStart(item)),
  };
};

export default connect(null, mapDispatchToProps)(GifCard)