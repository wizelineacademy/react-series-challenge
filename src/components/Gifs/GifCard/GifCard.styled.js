import styled from 'styled-components'
import { Favorite } from 'styled-icons/material'

const GifCardWrapper = styled.article`
  width: 248px;
  position: relative;
`

const GifCardItem = styled.a`
  height: 100%;
  width: 100%;
`

const GifImage = styled.img`
  width: 100%;
`

const GifCardFavorites = styled.div`
  display: flex;
  justify-content: flex-end;
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
`

const FavoriteLogo = styled(Favorite)`
  color: ${props => (props.isFavorite ? "red" : "white")};
  cursor: pointer;
  transition: all 0.3s ease;
  transform: scale(1);

  &:hover {
    transform: scale(1.1);
  }
`;

export {
  GifCardWrapper,
  GifCardItem,
  GifImage,
  GifCardFavorites,
  FavoriteLogo
}