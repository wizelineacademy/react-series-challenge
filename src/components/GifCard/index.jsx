import React from 'react';
import PropTypes from 'prop-types';
import {Btn, BtnsWrapper, GifCardWrapper} from "./GifCard.style";
import eyeIcon from './assets/eye.png';
import heartBorder from './assets/heart-border.svg';
import heartFilled from './assets/heart-filled.svg';
import {withRouter} from "react-router-dom";

export const GifCard = ({gif, history, isFavorite, isDetailBtnVisible, imageSize, toggleFavorite}) => {
  const image = imageSize === 'small' ? gif.images.small : gif.images.original;
  const heartIcon = isFavorite ? heartFilled : heartBorder;
  const openDetailView = () => history.push(`/detail/${gif.id}`);
  return (
    <GifCardWrapper>
      <img alt={'gif'} src={image} />
      <BtnsWrapper>
        {
          isDetailBtnVisible ? <Btn alt={'detail'} src={eyeIcon} onClick={openDetailView.bind(this)} /> : null
        }
        <Btn alt={'favorite'} src={heartIcon} onClick={toggleFavorite.bind(this, gif)} />
      </BtnsWrapper>
    </GifCardWrapper>
  )
};

GifCard.propTypes = {
  gif: PropTypes.shape({
    id: PropTypes.string,
    images: PropTypes.shape({
      original: PropTypes.string,
      small: PropTypes.string
    })
  }).isRequired,
  isFavorite: PropTypes.bool.isRequired,
  isDetailBtnVisible: PropTypes.bool.isRequired,
  imageSize: PropTypes.oneOf(['small', 'original']).isRequired,
  toggleFavorite: PropTypes.func.isRequired
};

export default withRouter(GifCard);
