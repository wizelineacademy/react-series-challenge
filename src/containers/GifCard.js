import GifCard from "../components/GifCard";
import {connect} from "react-redux";
import {toggleFavorite} from "../actions/favorite";

const mapDispatchToProps = (dispatch) => ({
  toggleFavorite: (gif) => dispatch(toggleFavorite(gif)),
});

const GifCardContainer = connect(
  null,
  mapDispatchToProps
)(GifCard);

export default GifCardContainer;
