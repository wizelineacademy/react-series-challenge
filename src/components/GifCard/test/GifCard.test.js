import React from 'react';
import GifCard from "../../../components/GifCard";
import {mount, shallow} from "enzyme";

const mockGif = {
  id: 'p3n7NWvfz3u0c1meay',
  images: {
    original: 'https://media3.giphy.com/media/p3n7NWvfz3u0c1meay/giphy.gif',
    small: 'https://media3.giphy.com/media/p3n7NWvfz3u0c1meay/200w_d.gif'
  }
};

describe('GifCard component', () => {
  it('should render', () => {
    const component = shallow(
      <GifCard.WrappedComponent
        gif={mockGif}
        isFavorite={false}
        isDetailBtnVisible={false}
        imageSize={'small'}
        toggleFavorite={() => {}}
      />
    );
    expect(component).toMatchSnapshot();
  });

  it('should not render visible btn', () => {
    const component = shallow(
      <GifCard.WrappedComponent
        gif={mockGif}
        isFavorite={false}
        isDetailBtnVisible
        imageSize={'small'}
        toggleFavorite={() => {}}
      />
    );
    expect(component).toMatchSnapshot();
  });

  it('should render visible btn and click it', () => {
    const toggleFavorite = jest.fn();
    const push = jest.fn();
    const component = shallow(
      <GifCard.WrappedComponent
        gif={mockGif}
        isFavorite={false}
        isDetailBtnVisible
        imageSize={'small'}
        toggleFavorite={toggleFavorite}
        history={{push}}
      />
    );
    const detailBtn = component.find('Btn').get(0);
    mount(detailBtn).simulate('click');
    expect(mount(detailBtn).prop('alt')).toEqual('detail');

    const favoriteBtn = component.find('Btn').get(1);
    mount(favoriteBtn).simulate('click');
    expect(mount(favoriteBtn).prop('alt')).toEqual('favorite');
  });

  it('should render filled heart', () => {
    const component = shallow(
      <GifCard.WrappedComponent
        gif={mockGif}
        isFavorite
        isDetailBtnVisible
        imageSize={'small'}
        toggleFavorite={() => {}}
      />
    );
    expect(component).toMatchSnapshot();
  });

  it('should render the big image', () => {
    const component = shallow(
      <GifCard.WrappedComponent
        gif={mockGif}
        isFavorite
        isDetailBtnVisible
        imageSize={'original'}
        toggleFavorite={() => {}}
      />
    );
    expect(component).toMatchSnapshot();
  });
});
