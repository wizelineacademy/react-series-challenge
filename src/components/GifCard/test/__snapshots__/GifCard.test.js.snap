// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifCard component should not render visible btn 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    gif={
        Object {
            "id": "p3n7NWvfz3u0c1meay",
            "images": Object {
              "original": "https://media3.giphy.com/media/p3n7NWvfz3u0c1meay/giphy.gif",
              "small": "https://media3.giphy.com/media/p3n7NWvfz3u0c1meay/200w_d.gif",
            },
          }
    }
    imageSize="small"
    isDetailBtnVisible={true}
    isFavorite={false}
    toggleFavorite={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <img
          alt="gif"
          src="https://media3.giphy.com/media/p3n7NWvfz3u0c1meay/200w_d.gif"
/>,
        <styled.div>
          <Btn
                    alt="detail"
                    onClick={[Function]}
                    src="eye.png"
          />
          <Btn
                    alt="favorite"
                    onClick={[Function]}
                    src="heart-border.svg"
          />
</styled.div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "gif",
          "src": "https://media3.giphy.com/media/p3n7NWvfz3u0c1meay/200w_d.gif",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Btn
              alt="detail"
              onClick={[Function]}
              src="eye.png"
/>,
            <Btn
              alt="favorite"
              onClick={[Function]}
              src="heart-border.svg"
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "alt": "detail",
              "onClick": [Function],
              "src": "eye.png",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "alt": "favorite",
              "onClick": [Function],
              "src": "heart-border.svg",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <img
            alt="gif"
            src="https://media3.giphy.com/media/p3n7NWvfz3u0c1meay/200w_d.gif"
/>,
          <styled.div>
            <Btn
                        alt="detail"
                        onClick={[Function]}
                        src="eye.png"
            />
            <Btn
                        alt="favorite"
                        onClick={[Function]}
                        src="heart-border.svg"
            />
</styled.div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "gif",
            "src": "https://media3.giphy.com/media/p3n7NWvfz3u0c1meay/200w_d.gif",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Btn
                alt="detail"
                onClick={[Function]}
                src="eye.png"
/>,
              <Btn
                alt="favorite"
                onClick={[Function]}
                src="heart-border.svg"
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "alt": "detail",
                "onClick": [Function],
                "src": "eye.png",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "alt": "favorite",
                "onClick": [Function],
                "src": "heart-border.svg",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`GifCard component should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    gif={
        Object {
            "id": "p3n7NWvfz3u0c1meay",
            "images": Object {
              "original": "https://media3.giphy.com/media/p3n7NWvfz3u0c1meay/giphy.gif",
              "small": "https://media3.giphy.com/media/p3n7NWvfz3u0c1meay/200w_d.gif",
            },
          }
    }
    imageSize="small"
    isDetailBtnVisible={false}
    isFavorite={false}
    toggleFavorite={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <img
          alt="gif"
          src="https://media3.giphy.com/media/p3n7NWvfz3u0c1meay/200w_d.gif"
/>,
        <styled.div>
          <Btn
                    alt="favorite"
                    onClick={[Function]}
                    src="heart-border.svg"
          />
</styled.div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "gif",
          "src": "https://media3.giphy.com/media/p3n7NWvfz3u0c1meay/200w_d.gif",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            null,
            <Btn
              alt="favorite"
              onClick={[Function]}
              src="heart-border.svg"
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          null,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "alt": "favorite",
              "onClick": [Function],
              "src": "heart-border.svg",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <img
            alt="gif"
            src="https://media3.giphy.com/media/p3n7NWvfz3u0c1meay/200w_d.gif"
/>,
          <styled.div>
            <Btn
                        alt="favorite"
                        onClick={[Function]}
                        src="heart-border.svg"
            />
</styled.div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "gif",
            "src": "https://media3.giphy.com/media/p3n7NWvfz3u0c1meay/200w_d.gif",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              null,
              <Btn
                alt="favorite"
                onClick={[Function]}
                src="heart-border.svg"
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            null,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "alt": "favorite",
                "onClick": [Function],
                "src": "heart-border.svg",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`GifCard component should render filled heart 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    gif={
        Object {
            "id": "p3n7NWvfz3u0c1meay",
            "images": Object {
              "original": "https://media3.giphy.com/media/p3n7NWvfz3u0c1meay/giphy.gif",
              "small": "https://media3.giphy.com/media/p3n7NWvfz3u0c1meay/200w_d.gif",
            },
          }
    }
    imageSize="small"
    isDetailBtnVisible={true}
    isFavorite={true}
    toggleFavorite={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <img
          alt="gif"
          src="https://media3.giphy.com/media/p3n7NWvfz3u0c1meay/200w_d.gif"
/>,
        <styled.div>
          <Btn
                    alt="detail"
                    onClick={[Function]}
                    src="eye.png"
          />
          <Btn
                    alt="favorite"
                    onClick={[Function]}
                    src="heart-filled.svg"
          />
</styled.div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "gif",
          "src": "https://media3.giphy.com/media/p3n7NWvfz3u0c1meay/200w_d.gif",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Btn
              alt="detail"
              onClick={[Function]}
              src="eye.png"
/>,
            <Btn
              alt="favorite"
              onClick={[Function]}
              src="heart-filled.svg"
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "alt": "detail",
              "onClick": [Function],
              "src": "eye.png",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "alt": "favorite",
              "onClick": [Function],
              "src": "heart-filled.svg",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <img
            alt="gif"
            src="https://media3.giphy.com/media/p3n7NWvfz3u0c1meay/200w_d.gif"
/>,
          <styled.div>
            <Btn
                        alt="detail"
                        onClick={[Function]}
                        src="eye.png"
            />
            <Btn
                        alt="favorite"
                        onClick={[Function]}
                        src="heart-filled.svg"
            />
</styled.div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "gif",
            "src": "https://media3.giphy.com/media/p3n7NWvfz3u0c1meay/200w_d.gif",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Btn
                alt="detail"
                onClick={[Function]}
                src="eye.png"
/>,
              <Btn
                alt="favorite"
                onClick={[Function]}
                src="heart-filled.svg"
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "alt": "detail",
                "onClick": [Function],
                "src": "eye.png",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "alt": "favorite",
                "onClick": [Function],
                "src": "heart-filled.svg",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`GifCard component should render the big image 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    gif={
        Object {
            "id": "p3n7NWvfz3u0c1meay",
            "images": Object {
              "original": "https://media3.giphy.com/media/p3n7NWvfz3u0c1meay/giphy.gif",
              "small": "https://media3.giphy.com/media/p3n7NWvfz3u0c1meay/200w_d.gif",
            },
          }
    }
    imageSize="original"
    isDetailBtnVisible={true}
    isFavorite={true}
    toggleFavorite={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <img
          alt="gif"
          src="https://media3.giphy.com/media/p3n7NWvfz3u0c1meay/giphy.gif"
/>,
        <styled.div>
          <Btn
                    alt="detail"
                    onClick={[Function]}
                    src="eye.png"
          />
          <Btn
                    alt="favorite"
                    onClick={[Function]}
                    src="heart-filled.svg"
          />
</styled.div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "gif",
          "src": "https://media3.giphy.com/media/p3n7NWvfz3u0c1meay/giphy.gif",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Btn
              alt="detail"
              onClick={[Function]}
              src="eye.png"
/>,
            <Btn
              alt="favorite"
              onClick={[Function]}
              src="heart-filled.svg"
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "alt": "detail",
              "onClick": [Function],
              "src": "eye.png",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "alt": "favorite",
              "onClick": [Function],
              "src": "heart-filled.svg",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <img
            alt="gif"
            src="https://media3.giphy.com/media/p3n7NWvfz3u0c1meay/giphy.gif"
/>,
          <styled.div>
            <Btn
                        alt="detail"
                        onClick={[Function]}
                        src="eye.png"
            />
            <Btn
                        alt="favorite"
                        onClick={[Function]}
                        src="heart-filled.svg"
            />
</styled.div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "gif",
            "src": "https://media3.giphy.com/media/p3n7NWvfz3u0c1meay/giphy.gif",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Btn
                alt="detail"
                onClick={[Function]}
                src="eye.png"
/>,
              <Btn
                alt="favorite"
                onClick={[Function]}
                src="heart-filled.svg"
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "alt": "detail",
                "onClick": [Function],
                "src": "eye.png",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "alt": "favorite",
                "onClick": [Function],
                "src": "heart-filled.svg",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
